pool_lib/__init__.py

```diff
@@ -1093,27 +1115,29 @@ def get_log_path(log_name: str, log_version: str) -> str:
         return None
 
     @sync
-    def gc(self,
-           recurse: bool = True,
-           verbose: bool = True) -> None:
+    def gc(
+            self,
+            recurse: bool = True,
+            verbose: bool = True
+    ) -> None:
         """Garbage collect stale data from the pool's caches"""
 
-        whitelist: Set[Tuple[str, str]] = set()
+        whitelist: set[tuple[str, str]] = set()
         for stock in self.stocks:
             for path, versions in stock.sources:
                 name = basename(path)
                 whitelist |= set([(name, version) for version in versions])
 
-            whitelist |= set(
-                [parse_package_filename(basename(path))
-                 for path in stock.binaries]
-            )
+            whitelist |= set([
+                parse_package_filename(basename(path))
+                for path in stock.binaries
+            ])
```

Comment:
Pretty sure this can be simplified to

```python3
whitelist |= { parse_package_filename(basename(path)) for path in stock.binaries }
```

Works because while `{ k: v for ... }` is dictionary comprehension, `{v for ... }` is set comprehension. Same as how `{ k: v }` is a dict but `{ v }` is a set.

Reply:
recommendation applied in https://github.com/turnkeylinux/pool/pull/18/commits/4902d5bde58a61ddd0942c8dea9f9497b161258a